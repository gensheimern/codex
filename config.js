module.exports = {
	DB_HOST: process.env.DB_HOST || '159.65.115.221',
	DB_USER: process.env.DB_USER || 'root',
	DB_PASS: process.env.DB_PASS || 'teamcodex',
	DB_NAME: process.env.DB_NAME || 'lunch_planner',
	JWT_SECRET: process.env.JWT_SECRET || 'secret',
	MAIL_HOST: process.env.MAIL_HOST || '159.65.115.221',
	MAIL_USER: process.env.MAIL_USER || 'support@codex-team.de',
	MAIL_PASS: process.env.MAIL_PASS || 'teamcodex123',
};
