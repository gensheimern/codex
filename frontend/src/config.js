const config = {
	debug: process.env.REACT_APP_DEBUG === 'TRUE',
	apiPath: process.env.REACT_APP_API_PATH,
	wsPath: process.env.REACT_APP_WS_PATH,
	basePath: process.env.REACT_APP_BASE_PATH,
}

export default config;