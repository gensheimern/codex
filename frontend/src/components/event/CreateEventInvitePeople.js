import React from 'react';
import AutoComplete from 'material-ui/AutoComplete';
import config from '../../config';

export default class CreateEventInvitePeople extends React.Component {

	constructor(props){
		super(props);

		this.state = {
			usersList: [],
			usersListConfig: {
				text: 'textKey',
				value:'valueKey',
			},
			groups: [],
			inviteList: [],
			inviteImageList: [],
			value: '',
		}
	}

	componentDidMount() {
		this.loadUsers();
		this.loadGroups();
	}

	loadUsers(){
		fetch(`${config.apiPath}/user/`, {
			method: 'GET',
			headers: {
				'Content-Type': 'application/json',
				'X-Access-Token': localStorage.getItem('apiToken'),
			}
		})
		.then((res) => {
			if (!res.ok) {
				throw new Error("Request failed.");
				} else if (res.status !== 200) {
				throw new Error("Forbidden");
			}

			return res.json();
		})
		.then(res => {
			this.setState({
				usersListLong: res,
			});

			this.setState({
				usersList: res.map((user) => {
					const { firstName, name, image, id, email } = user;

					return {
						textKey: `${firstName} ${name}`,
						ValueImage: image,
						ValueKey: id,
						ValueEmail: email,
					};
				}),
			});
		})
		.catch((err) => {
			this.setState({
				error: 'An Error occured.',
			});
		});
	}

	loadGroups(){
	    fetch(config.apiPath + "/team/", {
	      method: 'GET',
	      headers: {
	        'Content-Type': 'application/json',
	        'X-Access-Token': localStorage.getItem('apiToken')
	      }
	    })
	    .then((res) => {
	      if (!res.ok) {
	        throw new Error("Request failed.");
	        } else if (res.status !== 200) {
	        throw new Error("Forbidden");
	      }
	      return res;
	    })
	    .then(res => res.json())
	    .then(res => {
	        res.forEach(userid => {
	          console.log(userid);
	          this.setState({groups:[...this.state.groups, {textKey: userid.name, ValueEmail: userid.id, ValueKey:"Group"}]})
	        })
	        this.setState({groups: this.state.groups.concat(this.state.usersList)})
	    })
	    .catch((err) => {
	      this.setState({
	        error: 'An Error occured.',
	      });
	    });
	  }


	onChange = (event) => {
		this.setState({
			value: event.target.value,
		});
	}

	newRequest = (chosenRequest, index) => {
		this.setState((prevState) => {
			return {
				inviteList: [
					...prevState.inviteList,
					chosenRequest.ValueKey,
				],
				inviteImageList: [
					...prevState.inviteImageList,
					{ ...chosenRequest }
				],
				value: '',
			}
		});

		this.props.people(this.state.inviteImageList);
	}

	render(){
		return(
			<AutoComplete
				id="CreateEventInvitePeopleAutocomplete"
				fullWidth={true}
				value={this.state.value}
				onChange={this.onChange}
				floatingLabelText="Invite People"
				filter={AutoComplete.fuzzyFilter}
				dataSource={this.state.groups}
				dataSourceConfig={this.state.usersListConfig}
				maxSearchResults={5}
				onNewRequest= {this.newRequest}
			/>
		);
	}
}
