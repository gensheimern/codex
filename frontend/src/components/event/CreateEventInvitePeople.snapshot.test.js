/**
 * @author:Stella Neser
 *  Komponente ist ein Snapshottest
 */
import React from 'react';
import renderer from'react-test-renderer';
import CreateEventInvitePeople from './CreateEventInvitePeople';
import config from '../../config';
import {MemoryRouter} from 'react-router-dom';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

const fetchMock = require('fetch-mock');

beforeEach(() => {
    window.localStorage = {getItem: () => ""};
    fetchMock.get( config.apiPath +'/user/',
    [{
        id:7,
        firstName: "Max",
        name: "Mustermann",
        email: "max.mustermann@gmail.com",
        image: "",
        me: true,
    
    }]);
});

afterEach(() => {
    fetchMock.restore();
   });

describe('CreateEventInvitePeople Snapshot', () => {
    test('renders', () =>{
        const component = renderer.create(
            <MemoryRouter>
                <MuiThemeProvider>
                    <CreateEventInvitePeople/>
                </MuiThemeProvider>
            </MemoryRouter>
        );
        let tree= component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});