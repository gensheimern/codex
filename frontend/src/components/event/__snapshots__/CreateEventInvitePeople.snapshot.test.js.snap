// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateEventInvitePeople Snapshot renders 1`] = `
<div>
  <div
    style={
      Object {
        "display": "inline-block",
        "muiPrepared": true,
        "position": "relative",
        "width": "100%",
      }
    }
  >
    <div
      className={undefined}
      style={
        Object {
          "backgroundColor": "transparent",
          "cursor": "auto",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": 16,
          "height": 72,
          "lineHeight": "24px",
          "muiPrepared": true,
          "position": "relative",
          "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "width": "100%",
        }
      }
    >
      <label
        className={undefined}
        htmlFor="undefined-undefined-InvitePeople-21622"
        onClick={undefined}
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.3)",
            "lineHeight": "22px",
            "muiPrepared": true,
            "pointerEvents": "none",
            "position": "absolute",
            "top": 38,
            "transform": "scale(1) translate(0, 0)",
            "transformOrigin": "left top",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "zIndex": 1,
          }
        }
      >
        Invite People
      </label>
      <input
        autoComplete="off"
        disabled={false}
        id="undefined-undefined-InvitePeople-21622"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        open={false}
        style={
          Object {
            "WebkitOpacity": 1,
            "WebkitTapHighlightColor": "rgba(0,0,0,0)",
            "appearance": "textfield",
            "backgroundColor": "rgba(0,0,0,0)",
            "border": "none",
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "cursor": "inherit",
            "font": "inherit",
            "height": "100%",
            "marginTop": 14,
            "muiPrepared": true,
            "outline": "none",
            "padding": 0,
            "position": "relative",
            "width": "100%",
          }
        }
        type="text"
        value=""
      />
      <div>
        <hr
          aria-hidden="true"
          style={
            Object {
              "borderBottomStyle": "solid",
              "borderBottomWidth": 1,
              "borderColor": "#e0e0e0",
              "borderLeft": "none",
              "borderRight": "none",
              "borderTop": "none",
              "bottom": 8,
              "boxSizing": "content-box",
              "margin": 0,
              "muiPrepared": true,
              "position": "absolute",
              "width": "100%",
            }
          }
        />
        <hr
          aria-hidden="true"
          style={
            Object {
              "borderBottomStyle": "solid",
              "borderBottomWidth": 2,
              "borderColor": "#00bcd4",
              "borderLeft": "none",
              "borderRight": "none",
              "borderTop": "none",
              "bottom": 8,
              "boxSizing": "content-box",
              "margin": 0,
              "muiPrepared": true,
              "position": "absolute",
              "transform": "scaleX(0)",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "width": "100%",
            }
          }
        />
      </div>
    </div>
    <div
      style={
        Object {
          "display": "none",
        }
      }
    />
  </div>
</div>
`;
